src/Illuminate/Queue/Worker.php

```diff
@@ -304,12 +304,12 @@ protected function pauseWorker(WorkerOptions $options, $lastRestart)
     protected function stopIfNecessary(WorkerOptions $options, $lastRestart, $startTime = 0, $jobsProcessed = 0, $job = null)
     {
         return match (true) {
-            $this->shouldQuit => static::EXIT_SUCCESS,
-            $this->memoryExceeded($options->memory) => static::EXIT_MEMORY_LIMIT,
-            $this->queueShouldRestart($lastRestart) => static::EXIT_SUCCESS,
-            $options->stopWhenEmpty && is_null($job) => static::EXIT_SUCCESS,
-            $options->maxTime && hrtime(true) / 1e9 - $startTime >= $options->maxTime => static::EXIT_SUCCESS,
-            $options->maxJobs && $jobsProcessed >= $options->maxJobs => static::EXIT_SUCCESS,
+            $this->shouldQuit => self::EXIT_SUCCESS,
+            $this->memoryExceeded($options->memory) => self::EXIT_MEMORY_LIMIT,
+            $this->queueShouldRestart($lastRestart) => self::EXIT_SUCCESS,
+            $options->stopWhenEmpty && is_null($job) => self::EXIT_SUCCESS,
+            $options->maxTime && hrtime(true) / 1e9 - $startTime >= $options->maxTime => self::EXIT_SUCCESS,
+            $options->maxJobs && $jobsProcessed >= $options->maxJobs => self::EXIT_SUCCESS,
```

Comment:
Could this also be a leftover? Or is the change from `static` to `self` intentionally left as a part of this PR?

Reply:
While reviewing this file, I also felt compelled to say this change was unnecessary.

But after some thinking, and considering your previous remark regarding when enums should be used versus when constants should be used, I thought that, even if the class isn't final, there is no point on reading these constants from subclasses, as they are basically hard-coded exit values from processes, and should not change.

Using `self::` instead of `static::`, improves -- IMO -- intention declaration and correctness in this case.